src/GraphQL.Conventions/Adapters/Engine/GraphQLEngine.cs

```diff
@@ -76,8 +76,9 @@ public object Resolve(System.Reflection.TypeInfo typeInfo)
             }
         }
 
-        public GraphQLEngine(Func<System.Type, object> typeResolutionDelegate = null, ITypeResolver typeResolver = null)
+        public GraphQLEngine(Func<System.Type, object> typeResolutionDelegate = null, ITypeResolver typeResolver = null, IDocumentExecuter documentExecuter = null)
         {
+            _documentExecutor = documentExecuter ?? _documentExecutor;
```

Comment:
Now there is no reason to always create executor at [line 32](https://github.com/graphql-dotnet/conventions/blob/master/src/GraphQL.Conventions/Adapters/Engine/GraphQLEngine.cs#L32)

Reply:
Agreed, changing it